src/App.java

```diff
@@ -1,5 +1,562 @@
+         * Due to which the user can add more books here, however they can only add 5 books
+         * After which they have to re-run the program to add more books.
+         */
+        int numOfExtraSpace;
+        numOfExtraSpace= a; //A way to count down how much extra space the user has
+        if(a==0){//if the value is equal to 0
+            comms.consoleOutput("Sorry, allocated storage is full, restart the program to add more books");
+            menu();//return back to menu
+        }//end if statement
+        else{//If there is space to add more books
+        a--;//subtract 1 from this, to mark the reduction in space
+        comms.consoleOutput("In order to add a book you must answer the following questions: (Be carefull, if you mess up you have to restart the program)");
+        comms.consoleOutput("Note: You can only add "+numOfExtraSpace+ " additional books, to add more you must restart the program");
+        comms.consoleOutput("");//dynamic statement that displays the number of answers
+        comms.consoleOutput("What is the name of the book");//prompts user to add information about the book
+        title= comms.textConsoleInput();
+        comms.consoleOutput("");
+        comms.consoleOutput("What is the name of the author");
+        author= comms.textConsoleInput();
+        comms.consoleOutput("");
+        comms.consoleOutput("Is it borrowed currently, type in yes or no");
+        borrowed= comms.textConsoleInput();
+
+        if(borrowed.equalsIgnoreCase("yes")){
+            borrowedYN= true;//new borrowed boolean because the previous one can only turn the string values true/false to yes or no
+        }else if(borrowed.equalsIgnoreCase("no")){
+            borrowedYN= false;
+        }else{
+            comms.consoleOutput("Retry");
+            addBook();
+        }
+
+        if(borrowedYN==true){//Section of code only runs if book added is borrowed
+        comms.consoleOutput("");
+        comms.consoleOutput("Since it is borrowed, type in the borrower's name");
+        borrowerName= comms.textConsoleInput();
+        }else{
+            comms.consoleOutput("");
+            borrowerName= "N/A";
+        }
+
+        library.append(new Book(title, author, borrowerName, borrowedYN));//append method to add book to list
+
+        comms.consoleOutput("Your book has been added sending you back to menu");
+        booksToFile();//autosave
+        menu();//back to menu
+    }
+
+    }
+
+    public void removeBook(){//method to remove books
+        Book[] listOfBooks= library.getBooks();//list of book saved so that the method can use them
+        String bookName;
+        comms.consoleOutput("In order to remove a book type in its name");
+        bookName= comms.textConsoleInput(); 
+
+        for(int i=0; i<listOfBooks.length; i++){//loops through books to find the book that the user wants to delete

```

Comment:
this for loop should not be here - it should be inside the Library class in a `removeBook(int i)` method. You shouldn't be messing with the `Book[]` values at all in App.java - that should be fully abstracted as methods inside the `Library` class.